src/components/scenes/Staking/StakeOverviewScene.tsx

```diff
@@ -67,35 +68,41 @@ const StakeOverviewSceneComponent = (props: Props) => {
   const [rewardAllocations, setRewardAllocations] = React.useState<PositionAllocation[]>([])
   const [unstakedAllocations, setUnstakedAllocations] = React.useState<PositionAllocation[]>([])
 
-  // Background loop to force fetchStakePosition updates
-  const [updateCounter, setUpdateCounter] = React.useState<number>(0)
+  const updatePosition = React.useCallback(() => {
+    dispatch(updateStakingPosition(stakePlugin, stakePolicyId, wallet, account)).catch(err => {
+      console.error(err)
+      showError(err)
+    })
+  }, [dispatch, stakePlugin, stakePolicyId, wallet, account])
 
+  // Update the position on scene focus
+  const isFocused = useIsFocused()
+  React.useEffect(() => {
+    if (isFocused) {
+      updatePosition()
+    }
+  }, [isFocused, updatePosition])
+
+  // Update the position every minute
   React.useEffect(() => {
     const interval = setInterval(() => {
-      setUpdateCounter(updateCounter => updateCounter + 1)
-    }, 10 * 1000) // ten seconds
+      updatePosition()
```

Comment:
Never use use `setInterval` (in general), but *especially* when doing async work. There is no throttling, so tasks can pile up on top of each other and cause a big train-wreck.  Use `makePeriodicTask` instead, which correctly handles all the weird async race conditions.